File: Runtime/CoreLib.Script/Thread.js
///////////////////////////////////////////////////////////////////////////////
// Thread

var ss_Thread = function#? DEBUG Thread$##(start) {
	this.start = start;
	this.managedThreadId = 0;
    this.name = '';
};

ss_Thread.__typeName = 'ss.Thread';
ss.Thread = ss_Thread;
ss.initClass(ss_Thread, ss, {}, null, []);

ss_Thread.currentThread = new ss_Thread();

File: Runtime/CoreLib/Threading/Thread.cs
using System;
using System.Reflection;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace System.Threading
{
	[Imported(ObeysTypeSystem = true)]
	[ScriptNamespace("ss")]
	public class Thread {
		[IntrinsicProperty]
		public int ManagedThreadId { get { return 0; } }

		[IntrinsicProperty]
		public string Name { get; set; }

		[IntrinsicProperty]
		public static Thread CurrentThread { get { return null; } }

		public Thread() {
			
		}

		public Thread(ThreadStart start) {
		}

		public Thread(ParameterizedThreadStart start) {
		}
	}

	public delegate void ThreadStart();

	public delegate void ParameterizedThreadStart(object obj);
}